pycaret/containers/models/time_series.py

```diff
@@ -48,11 +48,27 @@
 from pycaret.utils.time_series import TSModelTypes
 from pycaret.utils.time_series.forecasting.models import _check_enforcements
 
+# First one in the list is the default ----
 ALL_ALLOWED_ENGINES: Dict[str, List[str]] = {
     "auto_arima": ["pmdarima", "statsforecast"]
 }
```

Comment:
can we somehow make this a part of the container itself? as an attribute. We can then access the allowed engines from the `models` attribute in the experiment object.

Reply:
I tried this but the problem is that there is a circular issue. Before I can set the models in the experiment, I need to set the engine and if I add the `allowed_engine` inside the container, I can not get that in the experiment before I set the models. 

Maybe we merge this now and if you have some time, you can have a look to if this can be improved? Let me know. Thanks!

Reply:
Ok, that sounds good!